pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -6805,3 +6805,232 @@ func TestDashbuilderWithPartialCORS(t *testing.T) {
 	assert.NotNil(t, cors)
 	checkCustomCors(t, cors)
 }
+
+func createStartupStrategy(strategyName string, ttl int) *api.StartupStrategy {
+	strategy := api.StartupStrategy{
+		StrategyName:               strategyName,
+		ControllerTemplateCacheTTL: Pint(ttl),
+	}
+	return &strategy
+}
+
+func TestKieServerOpenshiftStartupStrategyConfiguration(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			CommonConfig: api.CommonConfig{
+				StartupStrategy: createStartupStrategy(api.OpenshiftStartupStrategy, 4500),
+			},
+			Environment: api.RhpamProductionImmutable,
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{},
+			},
+		},
+	}
+
+	checkKiserverStartegyAssertions(t, cr, "4500")
+}
+
+func TestKieServerDefaultStartupStrategyConfiguration(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamProduction,
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{},
+			},
+		},
+	}
+
+	checkKiserverStartegyAssertions(t, cr, "5000")
+}
+
+func checkKiserverStartegyAssertions(t *testing.T, cr *api.KieApp, ttl string) {
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	envs := env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+	strategyEnabled, strategyEnabledTwo, strategyEnabledThree := checkEnvsKieserverStartupStrategy(t, envs, api.OpenshiftStartupStrategy, ttl)
+	assert.True(t, strategyEnabled)
+	assert.True(t, strategyEnabledTwo)
+	assert.True(t, strategyEnabledThree)
+}
+
+func checkEnvsKieserverStartupStrategy(t *testing.T, envs []corev1.EnvVar, strategy string, ttl string) (bool, bool, bool) {
+	strategyEnabled := false
```

Comment:
can we rename it? it is confusing, it seems to mean there are three different types of strategy.

Reply:
ok